GVRf/Framework/framework/src/main/java/org/gearvrf/GVRScene.java

```diff
@@ -556,7 +556,7 @@ public void onInit(GVRContext gvrContext, GVRScene scene) {
         private void recursivelySendOnInit(GVRSceneObject sceneObject) {
             getGVRContext().getEventManager().sendEvent(
                     sceneObject, ISceneObjectEvents.class, "onInit", getGVRContext(), sceneObject);
-            GVRScriptBehavior script = (GVRScriptBehavior) sceneObject.getComponent(GVRScriptBehavior.getComponentType());
+            GVRScriptBehaviorBase script = (GVRScriptBehaviorBase) sceneObject.getComponent(GVRScriptBehaviorBase.getComponentType());
```

Comment:
The scripting is not quite completely decoupled from the core framework due to all the support base classes. This PR is a great improvement and allows for footprint reduction. But if you'd like we could explore ways of decoupling completely later. For example, the scripting module could "listen" for the "onInit" event and internally handle it by propagating to scripting consumers - instead of the GVRScene class doing it.

Reply:
Agreed. The  listener idea is also a great suggestion. The main purpose of this change is for size reduction. Much more improvements in the scripting area need to be implemented.